app/models/graph_updater.rb

```diff
@@ -0,0 +1,53 @@
+module GraphUpdater
+  def self.call(graph, method, args)
+    case method
+    when 'preset_demand_setter'     then preset_demand_setter(graph, *args)
+    when 'demand_setter'            then demand_setter(graph, *args)
+    when 'share_setter'             then share_setter(graph, *args)
+    when 'conversion_setter'        then conversion_setter(graph, *args)
+    when 'reserved_fraction_setter' then reserved_fraction_setter(graph, *args)
+    when 'number_of_units_setter'   then set_unit(graph, *args)
+    else
+      raise ArgumentError, "no such method to update the graph #{ method }"
+    end
+  end
+
+  def self.preset_demand_setter(graph, node_name, value)
+    node = graph.converter(node_name)
+
+    node.preset_demand = value
+  end
+
+  def self.demand_setter(graph, node_name, value)
+    node = graph.converter(node_name)
+
+    node.demand = value
+  end
+
+  def self.share_setter(graph, edge_name, value)
+    link = graph.link(edge_name)
+
+    link.share = value
+  end
+
+  def self.conversion_setter(graph, slot_name, value)
+    node_name, carrier = slot_name.to_s.split('@')
+
+    node = graph.converter(node_name)
+    slot = node.slots.detect{ |s| s.carrier.key == carrier.to_sym }
```

Comment:
A small thing I noticed here: the format for conversion setters is `node_key@carrier`. However this assumes that the converter has only one matching slot when in fact it could have two: input and output.

If a node has an `electricity` input, and `electricity` and `loss` outputs, what does setting `node@electricity: 0.8` do? It could set either the input or output to 0.8. Atlas specifies slot keys as having a `+` or `-` after the node key, defining whether it is an input or output slot: `node_key[+-]@carrier`.

* `+`: Input slot, adds (+) energy to the node
* `-`: Output slot, takes away (-) energy from the node

For example:

```ruby
node = Atlas::Node.find(:households_flexibility_p2p_electricity)
(node.in_slots.to_a + node.out_slots.to_a).map(&:key)
```

* `households_flexibility_p2p_electricity+@electricity`
* `households_flexibility_p2p_electricity-@loss`
* `households_flexibility_p2p_electricity-@electricity`

Perhaps initializer methods could use the same format?